specs/Freeze.md

```diff
@@ -0,0 +1,176 @@
+browser renderer process and allows the operating system to reuse the memory that was
+used by the renderer process for other processes.   
+Note that the Suspend is best effort and considered completed successfully once the request
+is sent to browser renderer process. If there is a running script, the script will continue
+to run and the renderer process will be suspended after that script is done.
+See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+for conditions that might prevent WebView from being suspended. In those situations,
+The the completed handler will be invoked with isSuccessful as false and errorCode as S_OK.   
+The WebView will be automatically resumed when it becomes visible. Therefore, the
+app normally does not have to call Resume.
+The app can call `Resume` and then `TrySuspend` periodically for an invisible WebView so that
+the invisible WebView can sync up with latest data and the page ready to show fresh content
+when it becomes visible.   
+All WebView APIs can still be accessed when a WebView is suspended. Some APIs like Navigate will auto resume
+the WebView. To avoid unexpected auto resume, check `IsSuspended` property before calling APIs that might
+change WebView state.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```IDL
+interface ICoreWebView2_2 : ICoreWebView2 {
+
+  /// An app may call the `TrySuspend` API to have the WebView2 consume less memory.
+  /// This is useful when a Win32 app becomes invisible, or when a Universal Windows
+  /// Platform app is being suspended, during the suspended event handler before completing
+  /// the suspended event.
+  /// The CoreWebView2Controller's IsVisible property must be false when the API is called.
+  /// Otherwise, the API fails with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  /// Suspending is similar to putting a tab to sleep in the Edge browser. Suspending pauses
+  /// WebView script timers and animations, minimizes CPU usage for the associated
+  /// browser renderer process and allows the operating system to reuse the memory that was
+  /// used by the renderer process for other processes.
+  /// Note that the Suspend is best effort and considered completed successfully once the request
+  /// is sent to browser renderer process. If there is a running script, the script will continue
+  /// to run and the renderer process will be suspended after that script is done.
+  /// See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+  /// for conditions that might prevent WebView from being suspended. In those situations,
+  /// The the completed handler will be invoked with isSuccessful as false and errorCode as S_OK.
+  /// The WebView will be automatically resumed when it becomes visible. Therefore, the
+  /// app normally does not have to call Resume.
```

Comment:
Two questions:

1. Presumably `Resume` doesn't fail when `IsSuspended` is `false`, correct?
2. After the call to `TrySuspendAsync()` is awaited this condition will hold: `returnValue == IsSuspended`, correct?

Assuming the answer is yes to both, then I can't think of a case where the caller would need to handle the return value of `TrySuspendAsync` at all. At that point I'd just simplify the API to `SuspendAsync()`. Yes, it's a best effort but user code likely can't predict whether the rendered page supports it and wouldn't do anything differently anyway -- especially when user code typically doesn't even need to call `Resume`.

Reply:
You are right that TrySuspendAsync doesn't have to have a return value and the app can check IsSuspended value instead. It is a matter of whether the app has to check a property to check result or the async have a result to tell it the result. It feels that providing a result for an operation that naturally has a result is better.

#2 is not necessarily true with the current implementation. The current behavior is that the returned value only reflects the result of that operation. If the webview has been resumed before TrySuspendAsync completes, IsSuspended would be false while the operation is successful.